--- a/settings/Controller/UsersController.php
+++ b/settings/Controller/UsersController.php
@@ -422,7 +422,20 @@
 					if(empty($group)) {
 						$group = $this->groupManager->createGroup($groupName);
 					}
-					$group->addUser($user);
+					try {
+						$group->addUser($user);
+					} catch (Exception $e) {
+						$message = $exception->getMessage();
+						if (!$message) {
+							$message = $this->l10n->t('Unable to add user to group.');
+						}
+						return new DataResponse(
+							[
+								'message' => (string) $message,
+							],
+							Http::STATUS_FORBIDDEN
+						);
+					}
 				}
 			}
 			/**
