--- a/lib/public/appframework/http/contentsecuritypolicy.php
+++ b/lib/public/appframework/http/contentsecuritypolicy.php
@@ -78,7 +78,9 @@ class ContentSecurityPolicy extends EmptyContentSecurityPolicy {
 	/** @var array Domains from which object elements can be loaded */
 	protected $allowedObjectDomains = [];
 	/** @var array Domains from which iframes can be loaded */
-	protected $allowedFrameDomains = [];
+	protected $allowedFrameDomains = [
+		'\'self\'',
+	];
 	/** @var array Domains from which fonts can be loaded */
 	protected $allowedFontDomains = [
 		'\'self\'',
