--- a/lib/public/AppFramework/Http/ContentSecurityPolicy.php
+++ b/lib/public/AppFramework/Http/ContentSecurityPolicy.php
@@ -84,4 +84,12 @@
 	];
 	/** @var array Domains from which web-workers and nested browsing content can load elements */
 	protected $allowedChildSrcDomains = [];
+
+	public function __construct() {
+		$request = \OC::$server->getRequest();
+		// The host has already been checked against the trusted_domains in
+		// lib/base.php (function init).
+		$this->addAllowedConnectDomain('wss://' . $request->getInsecureServerHost());
+		$this->addAllowedFrameDomain('\'self\'');
+	}
 }
