--- a/core/Command/User/ResetPassword.php
+++ b/core/Command/User/ResetPassword.php
@@ -120,7 +120,7 @@
 
 
 		try {
-			$success = $user->setPassword($password);
+			$success = $user->setPassword($password, null, false); // Do not apply password policy
 		} catch (\Exception $e) {
 			$output->writeln('<error>' . $e->getMessage() . '</error>');
 			return 1;
--- a/lib/private/User/Database.php
+++ b/lib/private/User/Database.php
@@ -130,10 +130,12 @@
 	 *
 	 * Change the password of a user
 	 */
-	public function setPassword($uid, $password) {
+	public function setPassword($uid, $password, $validatePassword = true) {
 		if ($this->userExists($uid)) {
-			$event = new GenericEvent($password);
-			$this->eventDispatcher->dispatch('OCP\PasswordPolicy::validate', $event);
+			if ($validatePassword) {
+				$event = new GenericEvent($password);
+				$this->eventDispatcher->dispatch('OCP\PasswordPolicy::validate', $event);
+			}
 			$query = \OC_DB::prepare('UPDATE `*PREFIX*users` SET `password` = ? WHERE `uid` = ?');
 			$result = $query->execute(array(\OC::$server->getHasher()->hash($password), $uid));
 
--- a/lib/private/User/User.php
+++ b/lib/private/User/User.php
@@ -254,12 +254,12 @@
 	 * @param string $recoveryPassword for the encryption app to reset encryption keys
 	 * @return bool
 	 */
-	public function setPassword($password, $recoveryPassword = null) {
+	public function setPassword($password, $recoveryPassword = null, $validatePassword = true) {
 		if ($this->emitter) {
 			$this->emitter->emit('\OC\User', 'preSetPassword', array($this, $password, $recoveryPassword));
 		}
 		if ($this->backend->implementsActions(Backend::SET_PASSWORD)) {
-			$result = $this->backend->setPassword($this->uid, $password);
+			$result = $this->backend->setPassword($this->uid, $password, $validatePassword);
 			if ($this->emitter) {
 				$this->emitter->emit('\OC\User', 'postSetPassword', array($this, $password, $recoveryPassword));
 			}
