--- a/core/templates/installation.php
+++ b/core/templates/installation.php
@@ -8,7 +8,7 @@ script('core', [
 <input type='hidden' id='hasSQLite' value='<?php p($_['hasSQLite']) ?>'>
 <input type='hidden' id='hasPostgreSQL' value='<?php p($_['hasPostgreSQL']) ?>'>
 <input type='hidden' id='hasOracle' value='<?php p($_['hasOracle']) ?>'>
-<form action="index.php" method="post">
+<form action="index.php?redirect_url=/spreedbox-setup/" method="post">
 <input type="hidden" name="install" value="true">
 	<?php if(count($_['errors']) > 0): ?>
 	<fieldset class="warning">
