--- a/lib/private/Updater.php
+++ b/lib/private/Updater.php
@@ -307,6 +307,15 @@
 				$this->emit('\OC\Updater', 'finishedCheckCodeIntegrity');
 			}
 
+			// Re-enable disabled apps again
+			foreach ($disabledApps as $app) {
+				try {
+					\OC_App::enable($app);
+				} catch (\Exception $e) {
+					$this->log->logException($e, ['app' => $app]);
+				}
+			}
+
 			// only set the final version if everything went well
 			$this->config->setSystemValue('version', implode('.', \OCP\Util::getVersion()));
 			$this->config->setAppValue('core', 'vendor', $this->getVendor());
