--- a/lib/base.php
+++ b/lib/base.php
@@ -679,7 +679,9 @@
 		self::checkInstalled();
 
 		OC_Response::addSecurityHeaders();
-		if(self::$server->getRequest()->getServerProtocol() === 'https') {
+		$request = self::$server->getRequest();
+		if($request->getServerProtocol() === 'https' ||
+			(isset($request->server['spreedboxInternalRequest']) && $request->server['spreedboxInternalRequest'] === 'true')) {
 			ini_set('session.cookie_secure', true);
 		}
 
